### PR TITLE
Multilevel compression

### PR DESCRIPTION
**Pull request by nirvishek**
Completed issue #83 to generate multilevel compression file and upload the same to Glasswall s3.

Create Adhoc program to create multilevel zip files

Fetch one zip file with file with type other than zip from Minio or S3.
Create a zip of the file-1. Say _1.zip
Fetch another zip file with type other than zip from Minio or S3. file_2
Create a new zip with file_2 and _1.zip and name it as _2.zip
Run it in a loop to create and run it for 5 times, It will create 5 level zip files. Store all the files using Rabbit MQ job (Changed to direct storage to s3).
These zip file should be kept at a separate bucket "Multi_level_zip"

### CODE REVIEW

File: multilevel_compression/src/config.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class Config(object):
+    DEBUG = True
+    download_path = "/usr/src/app/download/"
+
+<<<<<<< HEAD
+    COMPRESSION_LEVEL = os.environ["COMPRESSION_DEPTH"]
+=======
+    COMPRESSION_LEVEL = 5
+>>>>>>> 25612eebb4d317bbb31e9c4280aff364ad30c300
```
Comment (DinisCruz) [inline multilevel_compression/src/config.py:16]: Merge conflict was added to the commit

**Review by DinisCruz — CHANGES_REQUESTED** [review]
See merge conflict issue

File: multilevel_compression/main.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        based on the level provided (level=5, requires 10 files) and
+        storing the resulting file back to target s3 bucket. """
+
+    def __init__(self):
+        self.s3 = S3Client(Config.S3_URL, Config.S3_ACCESS_KEY, Config.S3_SECRET_KEY)
+        logger.info("CreateMultilevelCompression::__init__ Creating s3 client object: %s" % self.s3)
+        self.level = Config.COMPRESSION_LEVEL
+        self.download_dir = Config.LOCAL_REPO_PATH
+
+    def get_files(self):
```
Comment (susilupwork) [inline multilevel_compression/main.py:22]: Good code

File: multilevel_compression/main.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            try:
+                if last_compressed_file:
+                    zip_obj.write(last_compressed_file)
+                zip_obj.write(file_1)
+                zip_obj.write(file_2)
+            except Exception as e:
+                logger.info("CreateMultilevelCompression::gen_multilevel_compression_file"
+                            "Got error {} while writing to the ZipFile '{}'".format(e, nzip_filename))
+            finally:
+                zip_obj.close()
```
Comment (susilupwork) [inline multilevel_compression/main.py:68]: Cool

**Review by susilupwork — APPROVED** [review]
Looks good.

**Review by susilupwork — APPROVED** [review]
Remove conflict with your own code.

File: multilevel_compression/src/config.py (hunk 1 above)
Comment (nirvishek) [reply multilevel_compression/src/config.py]: I have resolved the conflicted code.

**Review by susilupwork — APPROVED** [review]
Conflict merged.

**Review by dtollaku — APPROVED** [review]

**Pull request merged**